test/functional/change_stream.test.js

```diff
@@ -2038,17 +2036,19 @@ describe('Change Streams', function() {
 
         changeStream.next(() => {
           changeStream.next(() => {
-            changeStream.next(err => {
-              let _err = null;
-              try {
-                expect(err.message).to.equal('ChangeStream is closed');
-              } catch (e) {
-                _err = e;
-              } finally {
-                done(_err);
-              }
+            changeStream.close(err => {
```

Comment:
this doesn't seem to jive with the name of the test, it seems like it more wants something like a race between `close` and `next`, e.g.
```
changeStream.next(() => {
  changeStream.close(() => {
  });

  changeStream.next(err => {
    expect(err).to.exist; // ...
    done();
  });
});
```

Reply:
Yeah, I suspected that was the case, but then I don't really understand the reasoning behind this test. Without the change I made, the test flakes out because everything works; the change event gets processed before the close. However the test requires that the close happen before the change event - why? Wouldn't it be better it if the change event happens before the close?  I suspect I'm missing something here.